app/controllers/recipes_controller.rb

```diff
@@ -32,11 +33,17 @@ def update
       flash[:notice] = "You have successfully edit recipe!"
       redirect_to recipe_path(@recipe)
     else
-      flash[:alert] = "Error updating recipe!"
+      flash[:alert] = @recipe.errors.full_messages.first
       render :edit
     end
   end
 
+  def fork
+    @forked_recipe = @recipe.fork(current_user)
+    flash[:notice] = "Recipe was sucessfully forked"
+    redirect_to edit_recipe_path(@forked_recipe)
+  end
```

Comment:
I would suggest creating a `ForksController` with a `create` action to keep the whole thing RESTful. 

I would also take this a bit further and create a `Fork` class that would inherit from the Recipe class.

Reply:
Now it is RESTful, but after consideration will skip creating a `fork` model.